src/loading-screen/use-notify-bugsnag-on-timeout-status.ts

```diff
@@ -0,0 +1,14 @@
+import {useEffect} from 'react';
+import {LoadingState} from '@atb/loading-screen/types';
+import {notifyBugsnag} from '@atb/utils/bugsnag-utils';
+
+export const useNotifyBugsnagOnTimeoutStatus = (
+  status: LoadingState['status'],
+  paramsRef: LoadingState['paramsRef'],
+) => {
+  useEffect(() => {
+    if (status === 'timeout') {
+      notifyBugsnag('Loading boundary timeout', {...paramsRef.current});
```

Comment:
Perhaps the `'Loading boundary timeout'` string should be accepted as a parameter?

`useNotifyBugsnagOnTimeoutStatus` indicates that the function is general, but `'Loading boundary timeout'` is specific

Reply:
Good suggestion. I'll skip it for now since it as of now isn't a general hook, as it is tightly linked to loading both in typing, and that it isn't exposed out of the module through the index file.